List.cpp

```diff
@@ -0,0 +1,1038 @@
+    {
+        if (pos != end())
+        {
+            Node<T> *tmp = pos.ptr();
+
+            if (tmp->_prev != nullptr)
+            {
+                tmp->_prev->_next = tmp->_next;
+            }
+            else
+            {
+                _head = tmp->_next;
+            }
+
+            if (tmp->_next != nullptr)
+            {
+                tmp->_next->_prev = tmp->_prev;
+            }
+            else
+            {
+                _tail = tmp->_prev;
+            }
+
+            delete tmp;
+            _size--;
+        }
+    }
+
+    void resize(size_t size_n) { resize(size_n, T()); }
+
+    void resize(size_t count, const T &value)
+    {
+        if (_size < count)
+        {
+            for (size_t i = _size; i < count; i++)
+            {
+                push_back(value);
+            }
+        }
+        else
+        {
+            for (size_t i = _size; _size > count; i--)
+            {
+                pop_back();
+            }
+        }
+    }
+
+    void swap(Custom_List &other) noexcept
+    {
+        if (this == &other)
+        {
+            return;
+        }
+
+        std::swap(_head, other._head);
+        std::swap(_tail, other._tail);
+        std::swap(_size, other._size);
+    }
+
+    void merge(Custom_List &other)
+    {
+        if (this == &other)
+        {
+            return;
+        }
+
+        if (other.empty())
+        {
+            return;
+        }
+
+        if (empty())
+        {
+            splice(cbegin(), other);
+            return;
+        }
+
+        iterator it_begin(begin());
+        iterator it_end(end());
+        iterator other_begin(other.begin());
+        iterator other_end(other.end());
+
+        while (it_begin != it_end && other_begin != other_end)
+        {
+            if (*other_begin < *it_begin)
+            {
+                Node<T> *tmp(other_begin.ptr());
+
+                if (tmp->_next != nullptr)
+                {
+                    tmp->_next->_prev = tmp->_prev;
+                }
+                else
+                {
+                    if (tmp->_prev != nullptr)
+                        other._tail = tmp->_prev;
+                }
+
+                if (tmp->_prev != nullptr)
+                {
+                    tmp->_prev->_next = tmp->_next;
+                }
+                else
+                {
+                    if (tmp->_next != nullptr)
+                        other._head = tmp->_next;
+                }
+
+                if (it_begin.ptr() == _head)
+                {
+                    _head = tmp;
+                }
+
+                it_begin.ptr()->add(tmp);
+
+                ++_size;
+                --other._size;
+                ++other_begin;
+            }
+            else
+            {
+                ++it_begin;
+            }
+        }
+
+        splice(cend(), other);
+    }
+
+    void splice(const_iterator pos, Custom_List &other) noexcept
+    {
+        if (other.empty())
```

Comment:
Be very careful with `noexcept`. You cannot call not-noexcept methods inside `noexcept` marked one. Especially, writing something like a library/generic/common code. You must be strong in following the class interface.

If you use `noexcept` somewhere, you must be completely sure that the implementation does not use any method or function without `noexcept`.